src/components/MenuButton/MenuButton.tsx

```diff
@@ -326,49 +379,14 @@ const MenuButton: VibeComponent<MenuButtonProps> & {
         hideWhenReferenceHidden={hideWhenReferenceHidden}
         {...tooltipProps}
       >
-        <Dialog
-          wrapperClassName={dialogClassName}
-          position={dialogPosition}
-          containerSelector={dialogContainerSelector}
-          startingEdge={startingEdge}
-          animationType={AnimationType.EXPAND}
-          content={content}
-          moveBy={computedDialogOffset}
-          showTrigger={disabled ? EMPTY_ARRAY : DIALOG_SHOW_TRIGGER}
-          hideTrigger={hideTrigger}
-          showTriggerIgnoreClass={dialogShowTriggerIgnoreClass}
-          hideTriggerIgnoreClass={dialogHideTriggerIgnoreClass}
-          useDerivedStateFromProps={true}
-          onDialogDidShow={onDialogDidShow}
-          onDialogDidHide={onDialogDidHide}
-          zIndex={zIndex}
-          isOpen={isOpen}
-          hideWhenReferenceHidden={hideWhenReferenceHidden}
-        >
-          <TriggerElement
-            id={id}
-            data-testid={dataTestId || getTestId(ComponentDefaultTestId.MENU_BUTTON, id)}
-            type="button"
-            className={cx(styles.wrapper, overrideClassName, getStyle(styles, camelCase(`size-${size}`)), {
-              [styles.active]: isActive,
-              [getStyle(styles, openDialogComponentClassName)]: isOpen && openDialogComponentClassName,
-              [styles.disabled]: disabled,
-              [styles.text]: text
-            })}
-            aria-haspopup="true"
-            aria-expanded={isOpen}
-            aria-label={!text && ariaLabel}
-            onMouseUp={onMouseUp}
-            aria-disabled={disabled}
-            {...triggerElementProps}
-          >
-            {componentPosition === MenuButton.componentPositions.START && icon}
-            {text && <span className={styles.innerText}>{text}</span>}
-            {componentPosition === MenuButton.componentPositions.END && icon}
-          </TriggerElement>
-        </Dialog>
+        {tooltipChildren}
       </Tooltip>
     );
+
+    if (showTooltipOnlyOnTriggerElement) {
+      return DialogNode(TooltipNode(TriggerElementNode));
+    }
+    return TooltipNode(DialogNode(TriggerElementNode));
```

Comment:
I have to render DialogNode and TooltipNode as functions and not JSX because if they are rendered as JSX (which means they are not functions, but a functional component) Tooltip component (through Dialog's Refable) adds a span wrapper to it (which is the reason for this workaround at first place)

This should be the default at next major, even at cost of adding span by default